app/services/agreements/base_seeder.rb

```diff
@@ -0,0 +1,59 @@
+module Agreements
+  class BaseSeeder
```

Comment:
Possibly controversial, but I think this is maybe not a great case for inheritance because it's not always clear what needs to be overridden (the comment is great). Another option might be a builder class:

```ruby
module Agreements
  IaaStatusSeeder = SeederBuilder.build_class(
    filename: 'foo.yaml',
    record_class: FooRecord,
    primary_attribute_bundle: proc do |config|
      { 'foo' => config['foo']}
    end,
  ) do
    # optional block, that gets class_eval'd at the end of defining the class in case we need to override more
    def foobar
      # overrides something
    end
  end
end
```

In my mind, the advantages are:
- clear required params via keyword arguments makes it clear what needs to be provided

Disadvantages:
-  can be just as tricky to debug a dynamically defined class like this :shrug:

Reply:
I'll definitely look into it, thanks for the suggestion!

Reply:
Based on Slack chat w/ @zachmargolis we're deferring this for now, opening up a follow-up ticket for the future.